library/auth/src/main/java/com/epmedu/animeal/auth/AuthAPIImpl.kt

```diff
@@ -81,91 +77,94 @@ internal class AuthAPIImpl : AuthAPI {
         }
     }
 
-    override fun signUp(
+    override suspend fun signUp(
         phone: String,
-        password: String,
-        handler: AuthRequestHandler,
-    ) {
+        password: String
+    ): ApiResult<Unit> {
         val attrs = mapOf(
             AuthUserAttributeKey.phoneNumber() to phone,
         )
 
         val options = AuthSignUpOptions.builder()
             .userAttributes(attrs.map { AuthUserAttribute(it.key, it.value) })
             .build()
-
-        Amplify.Auth.signUp(
-            phone,
-            password,
-            options,
-            handler::onSuccess,
-            handler::onError,
-        )
+        return suspendCancellableCoroutine {
+            Amplify.Auth.signUp(
+                phone,
+                password,
+                options,
+                { resume(ApiResult.Success(Unit)) },
+                { resume(ApiResult.Failure(it)) }
+            )
+        }
     }
 
-    override fun signIn(
-        phoneNumber: String,
-        handler: AuthRequestHandler,
-    ) {
+    override suspend fun signIn(
+        phoneNumber: String
+    ): ApiResult<AuthSignInResult> {
         val authSignInOptions = AWSCognitoAuthSignInOptions.builder()
             .authFlowType(AuthFlowType.CUSTOM_AUTH)
             .build()
-
-        Amplify.Auth.signIn(
-            phoneNumber,
-            "",
-            authSignInOptions,
-            handler::onSuccess,
-            handler::onError,
-        )
+        return suspendCancellableCoroutine {
+            Amplify.Auth.signIn(
+                phoneNumber,
+                "",
+                authSignInOptions,
+                { resume(ApiResult.Success(it)) },

```

Comment:
Corresponding to this note of mine:
https://github.com/AnimealProject/animeal_android/pull/209/files#r1229578171
"AuthSignInResult" result is not currently used anywhere in the code, SignUpAndSignInUseCase treats this method as ApiResult\<Unit\>. I suggest to return here 
{ resume(ApiResult.Success(Unit)) },
and add comment above:
\\** Actual return type of onSuccess function here is AuthSignInResult, but currently it's unused *\
In an improbable case if we would actually need to use AuthSignInResult somewhere in the code, that would be a separate refactoring ticket which will change the structure

Reply:
done